clippy_lints/src/eta_reduction.rs

```diff
@@ -108,14 +108,20 @@ fn check_closure<'tcx>(cx: &LateContext<'tcx>, outer_receiver: Option<&Expr<'tcx
         {
             let vec_crate = if is_no_std_crate(cx) { "alloc" } else { "std" };
             // replace `|| vec![]` with `Vec::new`
-            span_lint_and_sugg(
+            span_lint_hir_and_then(
                 cx,
                 REDUNDANT_CLOSURE,
+                body.value.hir_id,
```

Comment:
Shouldn't this be `expr.hir_id` since that one is the actual closure, or do you intentionally want ppl to be able to do `|x| #[allow(...)] String::with_capacity(x)`?
Regardless, I don't think this would ever be an actual issue, just allows the attribute to be placed in even more nested ones than where it's semantically emitted, so 🤷